tests/network/services.go

```diff
@@ -247,67 +234,56 @@ var _ = SIGDescribe("Services", func() {
 		})
 
 		Context("with a service matching the vmi exposed", func() {
-			var jobsToCleanup []func() error
-			var serviceManager *vmiServiceManager
-			var serviceNamePrefix string
-
-			BeforeEach(func() {
-				serviceNamePrefix = "myservice"
-				serviceManager = newVMIServiceManager(inboundVMI, servicePort, serviceNamePrefix, selectorLabelKey, selectorLabelValue)
-
-				services := serviceManager.buildK8sServicesSpec()
-				Expect(services).NotTo(BeEmpty(), "a service should be exposed per each iface. At least one *must* be present.")
-
-				for _, exposedService := range services {
-					_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(&exposedService)
-					Expect(err).NotTo(HaveOccurred())
-				}
-
-				// reset the job cleanup functions list
-				jobsToCleanup = nil
-			})
+			var jobToCleanup func() error
+			var service *k8sv1.Service
 
 			AfterEach(func() {
-				var errors []error
-				for _, exposedService := range serviceManager.services {
-					err := virtClient.CoreV1().Services(inboundVMI.Namespace).Delete(exposedService.Name, &k8smetav1.DeleteOptions{})
-					errors = append(errors, err)
-				}
-				for _, err := range errors {
-					Expect(err).NotTo(HaveOccurred())
-				}
+				Expect(jobToCleanup).NotTo(BeNil(), "a k8sv1.Job cleaning up function should exist")
+				Expect(jobToCleanup()).To(Succeed(), "cleaning up the k8sv1.Job entity should have succeeded.")
+				jobToCleanup = nil
 			})
 
 			AfterEach(func() {
-				cleanupJobs(jobsToCleanup)
+				Expect(cleanupService(inboundVMI.GetNamespace(), service.Name)).To(Succeed(), "cleaning up the k8sv1.Service entity should have succeeded.")
 			})
 
-			It("should be able to reach the vmi based on labels specified on the vmi", func() {
-				for _, exposedService := range serviceManager.services {
-					jobCleanupFunc := assertConnectivityToService(exposedService.Name, inboundVMI.Namespace, servicePort)
-					jobsToCleanup = append(jobsToCleanup, jobCleanupFunc)
+			table.DescribeTable("should be able to reach the vmi based on labels specified on the vmi", func(ipFamily k8sv1.IPFamily) {
+				serviceName := "myservice"
+				if ipFamily == k8sv1.IPv6Protocol {
+					serviceName = serviceName + "v6"
+					service = buildIPv6ServiceSpec(serviceName, servicePort, servicePort, selectorLabelKey, selectorLabelValue)
+				} else {
+					service = buildServiceSpec(serviceName, servicePort, servicePort, selectorLabelKey, selectorLabelValue)
 				}
+
+				_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(service)
+				Expect(err).NotTo(HaveOccurred(), "the k8sv1.Service entity should have been created.")
+
+				jobToCleanup = assertConnectivityToService(serviceName, inboundVMI.Namespace, servicePort)
+			},
+				table.Entry("when the service is exposed by an IPv4 address.", k8sv1.IPv4Protocol),
+				table.Entry("when the service is exposed by an IPv6 address.", k8sv1.IPv6Protocol),
```

Comment:
The ipv6 test should be [skipped](https://github.com/kubevirt/kubevirt/pull/4156/files#diff-e05258dab4f2bda8138464ec283b1214R51) in case of a dual stack cluster.

Reply:
Done in https://github.com/kubevirt/kubevirt/pull/4224/commits/44a476d0338377bfc72e400ec2ad770fd2e7dca2 .

I'll rebase this if your PR gets in before this one, don't worry - I just didn't want to depend on it for something so simple.